### ISSUE TITLE
[Feature Request] Ability to use albumartist ID3 instead of artist

### ISSUE THREAD

**Issue by SaintDubious** [ampache/ampache#2799]
## Describe what you don't like about Ampache, or what it is missing

I've always been frustrated by the "artist" tag on an MP3 file, specifically when there's more then one artist on a song. When I try to list my collection by artist, I end up with a handful of entries for the same artist, it adds clutter. It's even worse when you have a couple of movie soundtracks and then suddenly the artist list contains a whole bunch of artists you've never heard of.

I'd like for there to be a feature that allows me to use the "albumartist" tag instead of the "artist" tag. This way I can keep the correct information on the file itself, but not deal with all the clutter when looking at my music collection

## Describe how you would like the feature to work**

I'd like for there to be a new boolean setting that allows me to use albumartist instead of artist when building a new catalogue. 

I actually already wrote all of this, but when I went to submit the PR I saw that I had written against the master branch, which is a totally different universe from the develop branch. Then I checked the CONTRIBUTING.md and saw that maybe I should have written this first. I'm happy to submit the PR against the 4.x build if that's possible. Please Advise.

**Comment by SaintDubious** [comment]
I noticed that other folks offer PR against master, so I went ahead and created the PR.

**Comment by lachlan-00** [comment]
i'm not sure if i'll be putting more change into 4.4 branches but there are a few removals in the database i need to do so master is fine.

**Comment by SePHTaN** [comment]
> and then suddenly the artist list contains a whole bunch of artists you've never heard of.
That's one thing i don't like also.
But my opinion is that all the data belonging to songs, albums and artists should go to the database without "malforming" them.
What you want to have is also achievable by not showing songartists in the view but album-artists.

There is always a way to show what you want from the data you have, but no way to show what you don't have.
It makes me a little bit scared seeing only the individual wishes of last months to tweak the contents of the database so it reflects what one wants to see, instead of making the database as extensive as possible and leave the rest to tweakable viewing abilities of ampache.

Why i am scared:
I just ran an update to my database cause i exchanged some of my albums with reencoded versions with higher bitrate.
Now i have several artists put together ("Die Toten Hosen" and "Die Roten Rosen" for example) that were correct separated before. All Albums of "The Toten Hosen" are now from "The Rote Rosen" except the one album i updated.
Also "The Toten Hosen" are missing musicbrainz-ids in the database i suspect having to do with an Feature Request just some weeks ago.
Update from Tags of the existing albums of The Toten Hosen (the Tags are totally correct just checked it) only say "Es wird keine Aktualisierung benötigt./ No actualisation needed" .

**Comment by SaintDubious** [comment]
Yeah, you have a fair point actually. It's usually better to change the *view* of the data, not the data itself. In this case there are 3 views I was considering, that get fairly complicated:

- The "Artist List View" which I'm trying to change, should show albumartist so that you don't see every permutation of the artist and all collaborators
- The "Sing List View" inside an album, which actually should show all of the songs and the *artist* field (including all collaborators)
- The "API View" which is how the API presents all of this to various clients.

Personally I actually only care about the "API View" (subsonic API specifically) because I use an android client against the ampache server. Sadly, as far as I can tell the Subsonic API only sends down a list of songs with a single artist field. So the data is lost and the client can only show one view of the artist. I tried to switch the API so that ampache sent the albmartist field instead, but then the client can actually make a request by artist ID and I thought this might cause issues if ampache thinks it's getting a request for artist, when the actual data is the albumartist.

All of this is to say that a true and correct fix is probably very hard, so I went for a band-aid. It looks to me like the 4.x line is drying up, as the "develop" branch is significantly different. It might not be worth pursuing this in 4.x. I'm actually really glad to hear both @lachlan-00 and @SePHTaN voice frustration with the current artist list (with all permutations of collaborators). I'm glad I'm not alone, maybe there can be a better way in 5.x?

Anyway, those are my thoughts. I'll leave the PR open for now in case there are other opinions, but I'm leaning towards closing it.

**Comment by lachlan-00** [comment]
There are more things available now and it'll get better over time too.
browse does default to album artist `browse.php?action=album_artist` and the groupings are a lot better.

We can close the pull but keep the issue and work on it a bit more

**Comment by lachlan-00** [comment]
tags are done a lot better these days this is probably done
